File: Demo/Objective-C_Storyboard/LGSideMenuControllerDemo/ChooseTableViewController.m
//
//  TableViewController.m
//  LGSideMenuControllerDemo
//

#import "ChooseTableViewController.h"
#import "MainViewController.h"

@interface ChooseTableViewController ()

@property (strong, nonatomic) NSArray *titlesArray;

@end

@implementation ChooseTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.titlesArray = @[@"Style \"Scale From Big\"",
                         @"Style \"Slide Above\"",
                         @"Style \"Slide Below\"",
                         @"Style \"Scale From Little\"",
                         @"Blurred root view cover",
                         @"Blurred covers of side views",
                         @"Blurred backgrounds of side views",
                         @"Landscape is always visible",
                         @"Status bar is always visible",
                         @"Gesture area is full screen",
                         @"Concurrent touch actions",
                         @"Custom style example"];
}

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.titlesArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];

    cell.textLabel.text = self.titlesArray[indexPath.row];

    return cell;
}

#pragma mark - UITableViewDelegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    UINavigationController *navigationController = [storyboard instantiateViewControllerWithIdentifier:@"NavigationController"];

    if (indexPath.row == self.titlesArray.count - 2) {
        [navigationController setViewControllers:@[[storyboard instantiateViewControllerWithIdentifier:@"OtherViewController"]]];
    }
    else {
        [navigationController setViewControllers:@[[storyboard instantiateViewControllerWithIdentifier:@"ViewController"]]];
    }

    MainViewController *mainViewController = [storyboard instantiateInitialViewController];
    mainViewController.rootViewController = navigationController;
    [mainViewController setupWithType:indexPath.row];

    UIWindow *window = UIApplication.sharedApplication.delegate.window;
    window.rootViewController = mainViewController;

    [UIView transitionWithView:window
                      duration:0.3
                       options:UIViewAnimationOptionTransitionCrossDissolve
                    animations:nil
                    completion:nil];
}

@end

File: Demo/Swift_Storyboard/LGSideMenuControllerDemo/ChooseTableViewController.swift
//
//  TableViewController.swift
//  LGSideMenuControllerDemo
//

class ChooseTableViewController: UITableViewController {

    private let titlesArray = ["Style \"Scale From Big\"",
                               "Style \"Slide Above\"",
                               "Style \"Slide Below\"",
                               "Style \"Scale From Little\"",
                               "Blurred root view cover",
                               "Blurred covers of side views",
                               "Blurred backgrounds of side views",
                               "Landscape is always visible",
                               "Status bar is always visible",
                               "Gesture area is full screen",
                               "Concurrent touch actions",
                               "Custom style example"]

    // MARK: - UITableViewDataSource
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return titlesArray.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)

        cell.textLabel!.font = UIFont.systemFont(ofSize: 16.0)
        cell.textLabel!.text = titlesArray[indexPath.row]
        
        return cell
    }
    
    // MARK: - UITableViewDelegate
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let storyboard = UIStoryboard(name: "Main", bundle: nil)
        guard let navigationController = storyboard.instantiateViewController(withIdentifier: "NavigationController") as? UINavigationController else { return }

        if (indexPath.row == self.titlesArray.count - 2) {
            navigationController.setViewControllers([storyboard.instantiateViewController(withIdentifier: "OtherViewController")], animated: false)
        }
        else {
            navigationController.setViewControllers([storyboard.instantiateViewController(withIdentifier: "ViewController")], animated: false)
        }
        
        guard let mainViewController = storyboard.instantiateInitialViewController() as? MainViewController else { return }

        mainViewController.rootViewController = navigationController
        mainViewController.setup(type: DemoType(rawValue: indexPath.row)!)

        let window = UIApplication.shared.delegate!.window!!
        window.rootViewController = mainViewController

        UIView.transition(with: window, duration: 0.3, options: [.transitionCrossDissolve], animations: nil, completion: nil)
    }

}
